tfhe/src/integer/server_key/radix_parallel/tests.rs

```diff
@@ -914,6 +932,114 @@ where
+        clear_0 %= modulus;
+
+        let (q_res, r_res) = sks.smart_div_rem_parallelized(&mut ctxt_0, &mut ctxt_1);
+        let q: u64 = cks.decrypt(&q_res);
+        let r: u64 = cks.decrypt(&r_res);
+
+        assert_eq!(clear_0 / clear_1, q);
+        assert_eq!(clear_0 % clear_1, r);
+
+        // Test individual div/rem to check they are correctly bound
+        let q_res = sks.smart_div_parallelized(&mut ctxt_0, &mut ctxt_1);
+        let q: u64 = cks.decrypt(&q_res);
+        assert_eq!(clear_0 / clear_1, q);
+
+        let r_res = sks.smart_rem_parallelized(&mut ctxt_0, &mut ctxt_1);
+        let r: u64 = cks.decrypt(&r_res);
+        assert_eq!(clear_0 % clear_1, r);
+    }
+}
+
+fn integer_default_div_rem<P>(param: P)
+where
+    P: Into<PBSParameters>,
+{
+    let (cks, mut sks) = KEY_CACHE.get_from_params(param);
+    let cks = RadixClientKey::from((cks, NB_CTXT));
+    sks.set_deterministic_pbs_execution(true);
+
+    //RNG
+    let mut rng = rand::thread_rng();
+
+    // message_modulus^vec_length
+    let modulus = cks.parameters().message_modulus().0.pow(NB_CTXT as u32) as u64;
+
+    for _ in 0..NB_TEST_SMALLER {
+        let mut clear_0 = rng.gen::<u64>() % modulus;
+        let clear_1 = rng.gen_range(1..modulus); // avoid division by zero
+        let clear_2 = rng.gen::<u64>() % modulus;
+
+        let mut ctxt_0 = cks.encrypt(clear_0);
+        let ctxt_1 = cks.encrypt(clear_1);
+
+        // add to change degree
+        sks.unchecked_scalar_add_assign(&mut ctxt_0, clear_2);
+        clear_0 += clear_2;
+        clear_0 %= modulus;
+
+        let (q_res, r_res) = sks.div_rem_parallelized(&ctxt_0, &ctxt_1);
+        let q: u64 = cks.decrypt(&q_res);
+        let r: u64 = cks.decrypt(&r_res);
+        let q_res_serialized = bincode::serialize(&q_res).unwrap();
+        let r_res_serialized = bincode::serialize(&r_res).unwrap();
+
+        assert!(q_res.block_carries_are_empty());
+        assert!(r_res.block_carries_are_empty());
+        assert_eq!(clear_0 / clear_1, q);
+        assert_eq!(clear_0 % clear_1, r);
+
+        // Test individual div/rem to check they are correctly bound
+        let q_res = sks.div_parallelized(&ctxt_0, &ctxt_1);
+        let q: u64 = cks.decrypt(&q_res);
+        assert!(q_res.block_carries_are_empty());
+        assert_eq!(clear_0 / clear_1, q);
+
+        let r_res = sks.rem_parallelized(&ctxt_0, &ctxt_1);
+        let r: u64 = cks.decrypt(&r_res);
+        assert!(r_res.block_carries_are_empty());
+        assert_eq!(clear_0 % clear_1, r);
+
+        // Determinism checks
+        let q2_res_serialized = bincode::serialize(&q_res).unwrap();
+        let r2_res_serialized = bincode::serialize(&r_res).unwrap();
+        assert_eq!(
+            q2_res_serialized, q_res_serialized,
+            "Operation was not deterministic"
+        );
+        assert_eq!(
+            r2_res_serialized, r_res_serialized,
+            "Operation was not deterministic"
+        );
```

Comment:
any reason you wrote the determinism check this way ? we should be able to just assert_eq on the ciphertexts from both computations ? I think we just compared ciphertexts directly

Reply:
I have no idea why I wrote it this way 😅

Reply:
no worries you have done a lot of code lately :)